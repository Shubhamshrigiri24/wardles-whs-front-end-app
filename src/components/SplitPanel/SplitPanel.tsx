import React, { FC } from "react";
import {
  Grid,
  makeStyles,
  Theme,
  useMediaQuery,
  useTheme,
} from "@material-ui/core";

export const useStyles = makeStyles((theme: Theme) => ({
  container: {
    paddingRight: theme.spacing(2),
    paddingLeft: theme.spacing(2),
  },
  secondaryPanelBreaksUpwards: {
    order: 1,
    [theme.breakpoints.up("md")]: {
      order: 2,
    },
    [theme.breakpoints.down("sm")]: {
      marginBottom: theme.spacing(3),
    },
  },
  mainPanelBreaksDownwards: {
    order: 2,
    [theme.breakpoints.up("md")]: {
      order: 1,
    },
  },
}));

export interface SplitPanelProps {
  main: JSX.Element;
  secondary: JSX.Element;
}

const SplitPanel: FC<SplitPanelProps> = ({ main, secondary }) => {
  const classes = useStyles();
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("sm"));

  return (
    <Grid
      container
      spacing={isMobile ? 0 : 3}
      direction="row"
      className={classes.container}
    >
      <Grid item xs={12} md={8} className={classes.mainPanelBreaksDownwards}>
        {main}
      </Grid>
      <Grid item xs={12} md={4} className={classes.secondaryPanelBreaksUpwards}>
        {secondary}
      </Grid>
    </Grid>
  );
};

export default SplitPanel;
