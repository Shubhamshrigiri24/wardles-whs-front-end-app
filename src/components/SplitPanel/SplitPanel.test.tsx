import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import SplitPanel, { SplitPanelProps } from "./SplitPanel";

describe("<SplitPanel />", () => {
  it("should match the snapshot", () => {
    const props = makeProps();
    const { container } = render(
      <SplitPanel {...props}>
        <h1>HI</h1>
      </SplitPanel>
    );
    expect(container).toMatchSnapshot();
  });
});

function makeProps(): SplitPanelProps {
  return {
    main: <h1>Main</h1>,
    secondary: <h1>Secondary</h1>,
  };
}
