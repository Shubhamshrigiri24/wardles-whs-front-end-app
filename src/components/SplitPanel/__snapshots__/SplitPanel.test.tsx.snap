// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplitPanel /> should match the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-container-1 MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root makeStyles-mainPanelBreaksDownwards-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
    >
      <h1>
        Main
      </h1>
    </div>
    <div
      class="MuiGrid-root makeStyles-secondaryPanelBreaksUpwards-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    >
      <h1>
        Secondary
      </h1>
    </div>
  </div>
</div>
`;
